### PR TITLE
Checks for temporary martial arts on mind transfer

### LINKED ISSUE #28489: I box, I die, I box again.
Dying with boxing gloves being equipped/Getting cloned while boxing gloves are still on your body result in a cloned player's hands remaining in a perpetual state of boxing, unable to punch lethally or grab.

### PR DESCRIPTION
**Pull request by ghost**
:cl: That Really Good Soda Flavor
fix: Martial arts will no longer be transferred in cloning, etc. if they are temporary (i.e. wrestling, krav maga).
/:cl:

closes #28518 
Fixes #28489

### CODE REVIEW

**Comment by Xhuis** [comment]
This is probably better than my own PR; @Cyberboss, would you prefer this instead?

File: code/datums/mind.dm [hunk 1]
```diff
@@ -117,7 +117,10 @@
 		C.last_mind = src
 	transfer_antag_huds(hud_to_transfer)				//inherit the antag HUD
 	transfer_actions(new_character)
-
+	if(martial_art)
+		if(martial_art.base) //Is the martial art temporary?
+			martial_art.remove(new_character)
+		martial_art.teach(new_character)
```
Comment (Cyberboss) [inline code/datums/mind.dm]: Put this into a separate proc

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by Cyberboss** [comment]
@Xhuis yeah

File: code/datums/mind.dm [hunk 2]
```diff
@@ -1626,6 +1623,14 @@
 			spell_list -= S
 			qdel(S)
 
+/datum/mind/proc/transfer_martial_arts(mob/living/new_character)
+	if(!ishuman(new_character))
+		return
+	if(martial_art)
+		if(martial_art.base) //Is the martial art temporary?
+			martial_art.remove(new_character)
+		martial_art.teach(new_character)
```
Comment (Cyberboss) [inline code/datums/mind.dm]: need an else between 1631 and 1632

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/datums/mind.dm (hunk 2 above)
Comment (Really-Good-Soda-Flavor) [inline code/datums/mind.dm]: no

Comment (AnturK) [inline code/datums/mind.dm]: Why ?

File: code/datums/martial.dm [hunk 3]
```diff
@@ -90,6 +90,15 @@
...
-	H.mind.martial_art = base
+	if(base)
+		H.mind.martial_art = base
+	else
+		H.mind.martial_art = H.mind.default_martial_art
 	if(help_verb)
 		H.verbs -= help_verb
+	remove_act(H)
+	H.mind.martial_art.teach(H)
+	qdel(src)
```
Comment (Cyberboss) [inline code/datums/martial.dm]: There's no way src could be default_martial_art here right? Because there's only one of those ever created

Comment (Really-Good-Soda-Flavor) [inline code/datums/martial.dm]: Oof..I'm gonna have to rethink how teach works

**Pull request merged**